server/routes/country_ambassador.js

```diff
@@ -0,0 +1,448 @@
+    let alumniId = req.params.alumniId
+    let country = req.params.country
+    try {
+        if (!isCountryAmbassador(alumniId, country)) {
+            res.status(400).send('Alumnus does not have access as ambassador for ' + country);
+            return;
+        }
+        let alumni = await fetchAllAlumniWithAccessContexts(country)
+        res.status(200).send({'alumni': alumni})
+    } catch (e) {
+        console.log('ambassador/allAlumni error: ' + e);
+        res.status(500).send({'ambassador/allAlumni error' : e})
+    }
+});
+
+router.get('/allStudents/:alumniId/:country', passport.authenticate('jwt', {session: false}), async (req, res) => {
+    let alumniId = req.params.alumniId
+    let country = req.params.country
+    try {
+        if (!isCountryAmbassador(alumniId, country)) {
+            res.status(400).send('Alumnus does not have access as ambassador for ' + country);
+            return;
+        }
+        let students = await fetchAllStudentsWithAccessContexts(country)
+        res.status(200).send({'students': students})
+    } catch (e) {
+        console.log('ambassador/allStudents error: ' + e);
+        res.status(500).send({'ambassador/allStudents error' : e})
+    }
+});
+
+router.get('/allSchools/:alumniId/:country', passport.authenticate('jwt', {session: false}), async (req, res) => {
+    let alumniId = req.params.alumniId
+    let country = req.params.country
+    try {
+        if (!isCountryAmbassador(alumniId, country)) {
+            res.status(400).send('Alumnus does not have access as ambassador for ' + country);
+            return;
+        }
+        let dbData = await schoolSchema.find({country: country})
+        res.status(200).send({'schools': dbData})
+    } catch (e) {
+        console.log('ambassador/allSchools error: ' + e);
+        res.status(500).send({'ambassador/allSchools error' : e})
+    }
+});
+
+router.patch('/toggleApprove/:alumniId/:country', passport.authenticate('jwt', {session: false}), async (req, res) => {
+    let alumniId = req.params.alumniId
+    let country = req.params.country
+    let profileId = req.body.profileId;
+    let type = req.body.type;
+    try {
+        if (!isCountryAmbassador(alumniId, country)) {
+            res.status(400).send('Alumnus does not have access as ambassador for ' + country);
+            return;
+        }
+        let dbData = []
+        let newApprovalState = false
+        let email = '', name = ''
+        if (type === 'ALUMNI') {
+            let alumni = await alumniSchema.findById(profileId);
+            newApprovalState = !alumni.approved
+            alumni.approved = newApprovalState;
+            let userRecordForAlumnus = await userSchema.findById(alumni.user, {email: 1})
+            email = userRecordForAlumnus.email
+            name = alumni.name
+            await alumni.save()
+            dbData = await fetchAllAlumniWithAccessContexts(country)
+        } else if (type === 'STUDENT') {
```

Comment:
These two cases can be put into a small function and unified

Reply:
noted, will change!

Reply:
addressed!